File: DSA/pranavdixit_590014418/Day_75_Question_2.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node *left, *right;
};

struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->left = node->right = NULL;
    return node;
}

void rightViewUtil(struct Node* root, int level, int* maxLevel) {
    if (root == NULL) return;

    if (*maxLevel < level) {
        printf("%d ", root->data);
        *maxLevel = level;
    }

    rightViewUtil(root->right, level + 1, maxLevel);
    rightViewUtil(root->left, level + 1, maxLevel);
}

void rightView(struct Node* root) {
    int maxLevel = 0;
    rightViewUtil(root, 1, &maxLevel);
}

int main() {
    struct Node* root = newNode(1);
    root->left = newNode(2);
    root->right = newNode(3);
    root->left->left = newNode(4);

    printf("Right View: ");
    rightView(root);
    return 0;
}

File: DSA/Animesh_jha_590016813/day_75_question_2.py
def right_view(root):
    if not root:
        return []
    
    result = []
    q = deque([root])
    
    while q:
        level_size = len(q)

        for i in range(level_size):
            node = q.popleft()
            if i == level_size - 1:
                result.append(node.data)
            
            if node.left:
                q.append(node.left)
            if node.right:
                q.append(node.right)
    
    return result

root2 = Node(1)
root2.left = Node(2)
root2.right = Node(3)
root2.left.left = Node(4)

print("Right View:", right_view(root2)) 
